### PR TITLE
Refactor Atoms

### PR DESCRIPTION
**Pull request by A-ryan-Kalra**
I've refactored all the individual atom variables into a completely new atom object

https://github.com/NayamAmarshe/writedown/assets/80680671/b9551eff-25d0-4626-bb99-fe80a82c2f8c

## Summary by CodeRabbit

- **Refactor**
	- Consolidated multiple note-related properties into a single `selectedNoteAtom` to simplify note management across the application.
- **New Features**
	- Enhanced note editing features by integrating all note properties (title, content, public status, and last updated time) into a unified interface, improving user experience and interaction efficiency.

### CODE REVIEW

File: components/dashboard/Sidebar/index.tsx [hunk 1]
```diff
@@ -37,7 +37,8 @@ const Sidebar = ({
   const [mounted, setMounted] = useState(false);
   const [createPostLoading, setCreatePostLoading] = useState(false);
 
-  const [postPublic, setPostPublic] = useAtom(postPublicAtom);
+  const [postPublic, setPostPublic] = useAtom(selectedNoteType);
```
Comment (NayamAmarshe) [inline components/dashboard/Sidebar/index.tsx:40]: This is incorrect. It should be called `selectedNote`, not `postPublic`. The value is `selectedNote.isPublic`.

File: stores/postDataAtom.ts [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
-export const postLastUpdatedAtom = atom<number | null>(null);
+type selectedNoteType = {
+  id?: string;
+  title?: string;
+  isPublic?: boolean;
+  content?: string;
+  lastUpdated?: number | null;
+};
+
+export const selectedNoteType = atom<selectedNoteType>({
```
Comment (NayamAmarshe) [inline stores/postDataAtom.ts:11]: The naming scheme is not correct. 
```suggestion
export const selectedNoteAtom = atom<SelectedNote>({
```

File: stores/postDataAtom.ts [hunk 3]
```diff
@@ -1,6 +1,17 @@
 import { atom } from "jotai";
 
-export const postTitleAtom = atom<string>("");
-export const postPublicAtom = atom<boolean>(false);
-export const postContentAtom = atom<string>("");
-export const postLastUpdatedAtom = atom<number | null>(null);
+type selectedNoteType = {
```
Comment (NayamAmarshe) [inline stores/postDataAtom.ts:3]: Naming scheme is not correct. Types should always be in Pascal case and there's no need to mention that it's a type.

```suggestion
type SelectedNote = {
```

**Comment by A-ryan-Kalra** [comment]
> Owner

Thanks for highlighting the issues. I've corrected them

File: stores/postDataAtom.ts (hunk 3 above)
Comment (NayamAmarshe) [reply stores/postDataAtom.ts:3]: @coderabbit Change the name of selectedNoteType to SelectedNote.

Comment (NayamAmarshe) [reply stores/postDataAtom.ts:3]: @coderabbitai Change the name of selectedNoteType to SelectedNote.

**Pull request merged**